### ISSUE TITLE
Documentation should be clearer on use of Shamiko and DenyList

### ISSUE THREAD

**Issue by bengalih** [adrianmmiller/Hide-My-Applist-Guide#1]
I posted this @ XDA:
https://forum.xda-developers.com/t/hide-my-applist-a-brief-guide.4519731/post-88258351

I think it would be beneficial if this guide was clearer in how the Magisk DenyList must be used to fully hide root from applications in conjunction with HMA.

While Enforce DenyList must be off in order for Shamiko to work, the apps which you want to hide root from must _STILL_ be listed inside the DenyList.

The way the guide is currently written it does not seem to reference this at all and therefore users trying to hide root and other apps using HMA may still see root detected if the app isn't listed on the DenyList.

**Comment by adrianmmiller** [comment]
I'm out and about at the moment but will look later, but initial response
is that people generally by the time they get to using HMA they should
already have knowledge of how to use deny list and shamiko. But I'm always
happy to take suggestions on improving things. Btw I no longer take part on
xda.....

**Comment by bengalih** [comment]
I agree for the most part.  I had all my stuff setup working no problem but haven't touched it for a while, so sometimes I forget specifically what is needed since I go through this whole process maybe once a year when I rebuild my phone.

In my case, an app update seemingly removed my app from DenyList and so it was detecting root.  In thinking this was an HMA problem I found your guide, and used it to rewalk through my setup.  As I mentioned on the XDA thread (which you can read even if you no longer post there) I'm still not sure why you use the Blacklist option instead of the Whitelist since the latter seems easier to configure and appears to work in more scenarios (at least for my use cases).

So, in fairness, I forgot to re-check DenyList.  I feel that if you are even going to mention Shamiko or DenyList being off then you should still reference the fact that the apps you are trying to hide root and other detection from should still be listed there.  Otherwise, you probably shouldn't even reference it at all since the instructions there seem incomplete.

That's just my 2 cents as I think it will help out others.  After all this is a beginner's guide and probably shouldn't assume they've done all the other steps properly.

**Comment by adrianmmiller** [comment]
Hi, sorry, have been sick the past few weeks and just starting to reocver, but will update the guide soon.

Yesterday i tried switching to Magisk Delta with SuList, and thought while im testing the alternatives, why not try HMA Whitelist....and then i saw there was no bulk way to set apps to Whitelist mode, so the number of clicks for my 270+ apps would have been well a lot more than the 2 clicks it takes me to restore my blacklist backup, and then the 3 to add my randomly named magisk app to it, and then reboot to have it all working.

I even messaged zgfg to see if i was missing something, but no. We are (theres more than one of us administering the guide now, on another github repo (not launched fully yet) set up entirely for documentation (and to avoid only one person having editing rights as per XDA threads)) looking at adding a whitelisting section (the stub is already there), but it looks a hard sell based on clicks alone, and as i suspect the author of HMA, like TopJohnWu with his advice for adding items to the MagiskHide/Deny List, the preference for using a blacklist model and sparingly adding things to it is to keep overheads down on processing. But i will reach out to the HMA author and see if theres an easy, or a way that could be implemented for those people who want to go down the whitelisting method road....

**Comment by bengalih** [comment]
Glad you're feeling better.

I think you are making somewhat of a false analogy regarding the effort to setup in Whitelist mode instead of Blacklist.

In your case, you already had setup a fully functioning Blacklist mode - so clearly the effort to set it up from scratch in Whitelist mode would have taken some time.

However, if you are starting from scratch you have two options:

1) Create a Blacklist template and then go into Select Apps and enable each app.  In this case you you would select "Enable Hide", leave the Work mode as default (Blacklist), and then select the template.  Then back out to the list. This would be 5 clicks per app once your template was defined (including selecting the app and backing out/save).  This is in effect what the guide states.

2) Go into Select Apps and enable each app.  In this case you you would select "Enable Hide", change the Work mode to Whitelist and then backout/save to the app list.  This is 3 clicks per app.

So effectively, the effort to setup Whitelist mode from scratch is 70% less effort than setting up Blacklist mode.
I think its really ridiculous to count clicks and apply math like this since in reality you are talking maybe a total of 3 minutes more to setup a list of a couple hundred apps, but since you bring it up I just wanted to point this out.

Again, yes, clearly it is quicker to restore your old configuration, but not for someone setting up new which is, I believe, the intent of the guide.  So, I don't get what you mean when you say its a "hard sell" for someone first setting up their HMA.  In fact, I would say when you know both options the Blacklist option is the hard sell based on work alone.  That doesn't even factor in that more apps seem to work better in Whitelist mode as I have found (and other users have chimed in on the XDA thread in agreement).

Also, as I mentioned in the thread it also means you don't need to constantly maintain a Blacklist template when you add new Lsposed plugins and other "offensive" apps.

That being said, there could certainly be some more options added to HMA to make configuration easier, but I don't believe you will ever get around the fact that you need to manually select your list of apps individually to configure.  What they could do is design an App Group List similar to the template.  So you can create multiple templates and multiple groups of Apps and then easily apply a template and other chosen settings to a group of apps.  That would be the ideal design but I'm not sure how necessary it is since most people shouldn't need to set this up more than once (with the exception of adding new apps, or if you forget to backup before a wipe).

Finally, I can't speak to your comment on overhead.  You may be correct here.  I have no idea if processing a Whitelist configuration is more overhead than a Blacklist one.  Offhand I wouldn't think so.  After all, a Whitelist configuration is effectively saying:

"Don't show any apps to the enabled apps unless I've added them to its Whitelist (or WL template)"

Whereas a Blacklist configuration is effectively saying:

"Show all apps to the enabled app unless I've added them to a Blacklist (or BL template)"

I'm not sure programmatically or processing wise one has to be more intensive than the other.  I could argue that Whitelist mode could actually be implemented (if it isn't already) in a more efficient manner, but I would be surprised if you could measure the difference (and how you would even reliably do so).

If the HMA devs provide information that processing apps in Whitelist mode is discernably slower (and I mean in reality, and not theoretically... If we are talking about an app taking .003 more seconds to open that is not worth discussing even if we can measure it) then you may have a stronger argument.  Until that is proven however I don't believe it a reason to prefer one over the other on that assumption.  If it can be shown to be visibly disruptive using Whitelist over Blacklist then that would be a compelling argument to stay on Blacklist mode as the default and only use Whitelist when an app has issues.  This, of course can be done in any event and at the very least the guide should be discussing how to use Whitelist mode because some apps just won't work without it.

One last thing - I guess I still don't quite understand your use case where you would need to actually configure either mode for 270 apps.  I probably have 400+ apps installed on my phone.  I probably use 10 of them regularly and maybe another 10 any given month.  I have never seen the need to have to configure HMA (or any other similar functionality) for more than a handful of them which have a problem with root/etc.  If the issue is utilizing Blacklist mode to hide other apps for privacy reasons, I still can't see the need for configuring most apps.

In any event, setting up HMA should generally be a one time thing (or very occasional thing) and what we are nit-picking about here is whether or not it is going to take you 5-10 minutes to do or 8-13.  I think this is silly to even debate.  The *main* reason to use Whitelist mode is because problematic apps (in my experience) work better when you block their access to everything in Whitelist mode (with no applied template), rather than in Blacklist mode where you are blocking only a few. 

I hope I've made my points clear and that you can now see your prior post doesn't seem to make any sense (to me) in context of this.  If I am misunderstanding you, or you have additional points to add, please lets discuss as I truly believe that Whitelist method is superior in pretty much every respect and if it isn't I would like to understand why.

**Comment by bengalih** [comment]
Also, I just realized this issue was really about the guide being clearer on making sure DenyList in Magisk was still configured when using Shamiko since the guide doesn't even really mention it.  I had only in passing referenced the Whitelist mode as per the XDA thread, so perhaps we shouldn't be getting the two conflated.

However both deal with documentation, so perhaps it is ok.  I don't know who is administering these issues, but if required we can open another issue and continue the conversation there.  Either way I am not too concerned as I am not directly involved in the project and these were all just my observations/suggestions for improvement for those who actually need the guide (people new to setting up HMA).

I'm happy to offer suggestions, but whether they are implemented or not is ultimately meaningless to me since I am not the target audience.

**Comment by adrianmmiller** [comment]
There are 7 items in my root blacklist (mostly lsposed modules), theyre applied to 3 apps, banking/finance/security related

Im not sure how this is more work....how is that hard to set up?

I'll take that over 270+ multiple clicks in whitelist mode

Maintaning the blacklist is as simple as adding a new lsposed module etc to the template and rebooting...

Ive tried reaching out to the author via TG to ask about an easier way to implement whitelist mode, and well, he immediately closed the PM, a fact i shared with ipdev and pndwal a little earlier...feel free to confirm. I am trying to bring the option of whitelisting to people, and have taken steps to do this, but clearly with the author recommending blacklist mode, and the conventtional paradigm of myself and the senior group to use it in this way (again i reached out to zgfg to make sure i wasnt missing something or misunderstanding that you have to go into every app in whitelist mode), along with the demonstrated massive time vampire of having to go into every app (instead of targetting less than a handful for most people), again, its not goign to appear as a winning method i feel to many people. But we are happy to add the option.

As for your initial contact about explaining the deny list/shamiko, as i did point out its assumed that people know to add items to the deny list, and IF using shamiko, to disable Enforce Deny List. There is a point where you have to maintain focus on your own topic and assume people know that if you suggest a requirement , that they have that sorted before using your guide. Theres a poitn where you risk having to epxlain all the way back to how magisk works, and thast not what the guide was started for. Knowing how to hide items via MagiskHide or Deny List should be a funadamental before you get anywhere near HMA, so i feel pointing out that if you use Shamiko, then you have to disable Enforce Deny List is a fair reminder, considering its also pointed out in the XDA magisk threads often enough.....

I feel this is written on point as it is, but i will add that items first need to be added to the deny list BEFORE installing Shamiko/toggling Enforce Deny List off, i felt that this was self evident, but im willing to accept that it might not appear that way:

---

Additionally, native Magisk denylist should be swapped for proper root hiding:

-  If running Magisk official, Shamiko with Enforce Deny List toggled OFF from https://github.com/LSPosed/LSPosed.github.io/releases

-  If running Magisk Delta, check with https://forum.xda-developers.com/t/discussion-magisk-delta-another-unofficial-third-party-magisk-fork.4460555/ for best hiding method

---

I felt that opening line meant  "if youre already using native Deny List (aka youve already set it up), then use one of these instead"

I dont want to spend my life revising and expanding things because people cant do the fundamentals :)

But i will expand on that and make it crystal clear, set up Deny List, but then preferably use Shamiko (on official Magisk) and toggle Enforce Deny List off.

As for Delta, im leaving the link as is as at last count there were 2 different options there (MagiskHide or SuList), and again i dont want to be regurgitating and documenting other peoples work, which people should seek out and have a fundamental understanding of before they hit this guide, aka why all this is under the Requirements heading...

Not being a hardass on purpose about any of this as really before i made this guide, all anyone got was a few lines about HMA setup that confused people. I set about making it infinitely easier, and with pictures because even i found the barest description of how to set it up from zgfg, who was the main source for it, confusing at first...i got then progressively tired watching each new person enquire about it and be equally confused. Since i could read the writing on the wall that no one else was going to put up even halfway decent steps on how to configure it, i (somewhat reluctantly) decided to once and for all try and end the madness. Its infinitely more helpful than zgfg posting links to his earlier very short 3 line explanation....and being quite tired of anyone asking for more help

And i thought i had sorted things nicely, until you came along :)

At time of writing ill be honest with you, only you, and now, for reasons best left to him to address, zgfg has decided (after several years) to switch from using the blacklist method which he was the originator of and espoused to us all, as far as i know use whitelisting.  In fact i was surprised when he switched, and as i said to my cohorts, "it seems a whitelisitng page is now pretty much a  done deal, as for some reason now, after many years, zgfg has deciced to use whitelisting"

Personally for me thats a LOT more work, and isnt time effective in this IT persons busy life, when on a new flash (which i do every month, being a Pixel user (and often inbetween monthly stock updates a custom ROM tester)), all i have to do i restore my lsposed backup, my HMA one, go to the blacklist template and add the newly hidden random magisk app name, and reboot. I like low maintenance andf faffing about. I leave all the high maintenance stuff for my clients :)

Ill update you when i hope to have the changes made this weekend, you can feel to add comments/suggestions/corrections you have as issues on that new github, as there will be 3 eyes on that one.....and any decison made there is pro/conned by us 3.

**Comment by bengalih** [comment]
Ok - one of us is clearly on the wrong page and, I'll admit it might be me, so lets reset for a moment.

For starters, I'm not trying to make anyone's life harder, just the opposite.  I learned to use HMA on my own from just their github and maybe a paragraph I found elsewhere on Google.  I'm sure some people need the guide and I had just happened upon it since it had been 9 months since I first set it up and I was having some issues and was just googling around to see if anything had changed or if there were any new options since I just updated that might help with my problem.  As I mentioned, the error ended up being my own (to an extent) because the app in question had gotten removed from the Magisk DenyList (used with Shamiko).  I had clearly known to set this up, because I did it initially, but never thought to check back because:
a) I had no reason to know it had been removed and
b) I forgot somewhat about this step because maintaining these isn't something I do on a daily/weekly/monthly basis.  I do it once when I set my phone up and that's it.

I don't consider myself a pro/expert, but I have been modding phones since Palm Pilots (ok - that's technically not a phone, but you get the gist...plus I did have the modem attachment!).  This is why I started the initially simple request that I thought the guide should at least *mention* that you should ensure your apps are properly excluded on the DenyList since if I overlooked them a newbie very well might.

Anyway, going back to the whitelist/blacklist and where we seemingly have a disconnect.

You state above:

"There are 7 items in my root blacklist (mostly lsposed modules), theyre applied to 3 apps, banking/finance/security related

Im not sure how this is more work....how is that hard to set up?

I'll take that over 270+ multiple clicks in whitelist mode"

I would say in your words about my configuration:

There are 0 items in my root blacklist, since I didn't need to set one up.  I went into 5 apps, banking/finance/security related and ticked off Enable and set them to Whitelist mode.

I'm not sure how that level of work is substantially different than what you did?
I don't know where you are getting the idea that I need to go and enable Whitelist mode on every app?

Whitelist/Blacklist is an option on each individual app.  If you have no need for HMA to hide something from an app then you don't need to go into that app and enable HMA on it.

In other words, If I have 270 apps and only need to block my "offensive" apps from 3 of them then I only need to click on those 3 apps in my HMA list, click "enable" and then either leave the default Blacklist mode or toggle them to Whitelist.  In either case you can also chose to apply a template (which, as discussed is not usually needed in Whitelist mode, though almost always required in Blacklist).

Maybe you have a different understanding but I can't for the life of me think why you would go and click all all of your apps to do anything to them in HMA apart from the 3 you need HMA to operate on.

What am I missing?

**Comment by adrianmmiller** [comment]
directly from zgfg last night when i asked if i needed to enable whitelist of all apps, because i have 274 of them:

- No Whitelist templates needed for HMA
- You don't save clicks 
- For every app in App manage:

    - instead of enabling Hide and attaching Blaclist template

    - you need to enable Hide, switch to Whitelist mode and Exclude System apps (this last check is not needed for all but is safer to put for all than thinkering where you need or not)_

**Comment by bengalih** [comment]
I think you have misunderstood him.  When he says "for every app in app manage" he means for every app you want to manage with HMA, not every single app.

Why would you need to go into an app your don't "enable" in HMA to apply either blacklist of whitelist.

It seems you trust zgfg as he taught you all how to use it (by your own admission).
Perhaps you should think as to why he seems to agree with my suggestion of whitelist?

I'm happy to admit I'm incorrect if proven so, but I believe you don't yourself fully understand how the tool works.

I can tell you I am successfully using every app on my phone, including the handful of apps I have configured in HMA (banking/finance).  All I needed to do was enable each of those few apps in HMA and toggle "Whitelist" to on.  That's it.  Been running that way for 6-9 months.

**Comment by adrianmmiller** [comment]
I have aspergers, we are a very literal people...so when someone tells me every app, and theyve known me for years, i believe them...

If theres been an error in misunderstanding, im happy to be proven wrong :)

**Comment by adrianmmiller** [comment]
Okay, update:

Since your last i have rectified my misunderstanding and performed the whitelisting steps on JUST the apps i would normally have on my blacklist template, rebooted, then cleared data on my testing apps, and they pass

So i do finally get what you were trying to get across, with a layer of (un)helpful misunderstanding in the way, and my wonky brain :)

**Comment by adrianmmiller** [comment]
in case that last message wasnt enough, i wholly apologise for not getting your point a WHOLE lot earlier

**Comment by bengalih** [comment]
No worries at all, I knew there was a disconnect and was just trying to figure the best way to resolve.

I was still a bit confused because even if your way was necessary (i.e. enabling every app), while that would have been a PITA to do the first time it still shouldn't affect every reflash because a backup/restore of HMA should keep all that configuration(?).  I say that without ever having tried it myself, but that should be the intent of the backup configuration.

Anyway, so you understand my point now that configuring Whitelisting is not more work and, in actuality is probably less work for someone just setting up for the first time.

Additionally, as I know from my own experience (and at least a couple others agree in the thread), certain apps that refuse to work in Blacklist mode work fine in Whitelist mode.  This is why I use it - less hassle that something doesn't work.  I've also never seen an app that fails to work in Whitelist mode, but does in Blacklist...

My only concern, which you brought up initially, is that the devs of HMA do state (in the app) that Whitelist is "Recommended", though I have never seen anything published as to why.  You may be correct that there is more processing overhead, but I can't notice it, nor have I noticed any stability problems.

_On a tangent...FWIW, my personal experience is anyone whose main chat channel for their app is TG is generally going to be (on the average) an a-hole when it comes to dealing with users.  I find this doubly true (at least!) for the asian devs.  Clearly those guys are very bright, but they have zero ability to interface with anyone and I'm sure they care not one bit (if they even know) that this guide exists or about the community at XDA at all.  So your experience with being shut down when messaging them has nothing to do with the validity of the question it is due primarily to them simply not wanting to deal with anyone on anything and amplified probably that you do not speak their language (either literally as in Chinese, and the programming language.).  They only will interface with other devs who want to actually talk direct code or with whoever all the other people in that channel are who just like sending their silly emoji around constantly (I shouldn't judge as this is a cultural thing, but it makes even intelligent people look like 8 year olds).  Anyway, point being that their lack of response to you had nothing to do with the merit of your questions._

Give the above, I can't find a reason to actually honor the Devs' informational point that Blacklist mode is recommended, because there is no practical evidence it is better.  My only assumption is that *some* apps maybe do have a problem with WL mode and so the default is to "fail open" by ensuring that apps will run, even if they still don't properly block root, etc.

I of course welcome you and anyone else to operate in WL mode and get more feedback if you find issues with performance or stability compared to BL mode.  I just haven't seen any yet.

And, as I inferred before there is really nothing *wrong* with using BL mode as you have provided in the guide.  It is just a bit more setup (but negligible) and if someone did find an app that still detected root they could always go and turn on WL just for that app.  Really whatever works!

If I were writing the guide however I would present both setting up in BL and WL mode separately.  I.e. explaining what each of the modes actually means and giving an overview of the reasoning behind each.  Then I would recommend one over the other (you know my preference, but I am not about to force that on anyone).  Finally I would make it clear that you can choose to mix and match if one isn't working over the other.

This last fact I would make very clear, especially considering your confusion on the above.  Make it clear that apps only need to be Enabled if you have the need to hide something else on the phone from them and that you can selectively choose to obscure apps from them using either WL or BL on a case-by-case basis.

I think my work here is done, we are on the same page and I've given my suggestions.  You all are the ultimate arbitrators of the Guide and will do what you deem best (or least time consuming?).  I just hope that, in maintaining a guide focused on "newbies", you try to instill in them the correct information and the reasons for why so that there is no reason for them to have to question.

I probably won't even reference it myself again, unless of course someone can give me a good description of exactly what the "vold data" option does and when you might use it :)

**Comment by adrianmmiller** [comment]
Lol, thanks for bearing with me through my misunderstanding - i did theorise to pndwal and ipdev on our 3way TG group just now, in owning up to having borked understanding where you were coming from, that perhaps id had another mini stroke (on top of misunderstanding zgfg's brief instructions) :)  I had a seizure in 2017 and basically didnt get vertical for 3 years, and was just starting to get upright and mobile when of course COVID hit :)

I'll always own up to things if it turns out IM the issue :)

Interestingly the one app id hoped might have shown some possible difference in whitelist mode, my local travel app, still wont allow top ups in it, its not detecting root (it would baulk and refuse to launch if it did), but it still fc's when i hit top up mode...but its been that way for years both before and after HMA and blacklisting....id just had my fingers crossed that *maybe* something would change for that one, nope...

So, okay, you have my attention, and im on board....and well the documentation and screenshots for whitelist mode will be much shorter than my addled brain thought :)

You can see that whitelist was always going to be added, as the stub on the repo page at the new doco only repo for some popular modules has been there for about a week :)

https://i.imgur.com/kGm5eqb.jpeg

the other two have already edited the original format of my guide a bit, and tidied up the layout a bit (images in sparate folder for housecleanming, somethgin im not overly concerned about, im a function over form guy). So it does look different to the current version of the guide here on my perosnal github. Its currently set private ont he new repo, but should go live this weekend hopefully.

its just gone nowhere int he last week or so because ive been unwell and pndwal has been fiddling with the content (the other night saying "i didnt realise i had editing rights, i thought i was creating a pull request, feel free to revert :) )

eventually https://github.com/mModule will host a collection of more easily maintained documents/wikis for popular modules

im much more comfy at documenting stuff ive created as i understand things ive personally made or coded....i absolutely coming in on someone elses work, but the HMA guide was i felt worth doign the way it was origianlyl done, as several times a week people were getitng told to use it, but they had if they were lucky, a link to 3 lines of text that confused people


case in point about being more comfy doco'ing my own stuff, i know it makes no sense, but though its more complex than HMA, to me documenting my own code like this was far easier (my old repo):

https://github.com/stylemessiah/Unifi-Auto-Ban

yep, i can undertsand complex PHP arrays and create a script that creates all the necessary ancillary files, but HMA's blacklist vs whitelist paradigm owned me :)

as for the TG devs, yes a lot of the asian devs can be dismissive...not a great difference from the magisk dev's....all of us "senior" magisk related posters like pndwal, ipdev and even osmosis have experienced plenty of "spice" when creating issues ont he magisk github. i think i only had one win in all the year s i poste don there, having the magisk app stay on the modules tab after a local install (i always install modules locally from my private store)...that only took about 9 months from suggestion, and 2 elevations from osmosis. We recently decided to runt he gauntlet by askign for them to add native backup/restore fucntionality for suepruser and magisk hide/deny list (because currently, the only way to have those functions is to use my modules, and unsurprisingly they were quickly shot down, even with backing of the others. Osmosis probably cops more flack than anyone for trying to improve things, but for whateveer reason ToppJohnWu left magisk in the hands of a cabal who hates outsiders...despite their level and years of imvolvement....

so yeah, we know all about "dev spice"

the whtielist vs blacklist doco will show clear reaosns why you might do either, and now resonates far easier in one persons brain than mere hours ago :)

ill perhaps call on you to proofread things so you can fact check me (by inviatation this time lol)

vold data...in the most basic terms its just one way to try and futher isolate apps from detecting other apps....i have a module for appdata isolation (throw into my lap by stillhard, who asked me to package it up- afetr promising to a TG group id do to:) ) over here:

https://github.com/adrianmmiller/AppDataIsolation

Its a method that is targetted at A11+ (HuskyDG has his own variant for A11-), and it has one interesting side effect for me perosnally (and theres a warning on the page for this....if i try and run a Migrate backup (my chosen backup solution for years) without first toggling the module off, my backups do not include app data. But Vold isoaltion in HMA toggled on causes no such issue for me. As a result i havent used my own isolation module since i discovered the issue. I beleive others do not necessarily see issues with it, perhaps because they dont do backups ro use a different backup app.

Some background on what it does is included on the modules page, which may help

**Comment by bengalih** [comment]
Thanks.  You can ping me here or on XDA if you'd like me to take a look at anything in the future.

**Comment by adrianmmiller** [comment]
Just a heads up, the doco isnt up yet as i am having a terrible time with trying whitelist and my Watch 4 (which ive just tried to connect after March Pixel update and setting up whitelist the other day, i hadnt gotten around the Watch till today, too busy), which has worked flawlessly in blacklist mode for 2 years, not limited to:

Watch 4 connection getting severed on reboot - meaning you have to reset the watch and start over - i think you can see the circular hell this causes

Samsung Health setup completed fine, but when trying to setup Samsung Health Monitor (separate app but can send BP to Samsung Health) it wants to install Samsung Health, which  of course is installed and setup and so this is a circular hell (as well)

Have tried all permutations of settings (clearing data and reboots between) to overcome these issues, no effect. Note: Samsung apps are known to be hellish with root related matters....

Personally going back to blacklist mode, because i dont have time for this. Life was much simpler in blacklist mode :)

p.s. For some more context, when running through the just the Watch 4 setup wizard in whitelist mode, there were steps/screens missing....it was weird, never seen that before, so i was unsurprised when it severed its connection on reboot

**Comment by adrianmmiller** [comment]
Can you ping me at adrian.m.miller@gmail.com thanks when you have a minute so i can run some things by you

**Comment by adrianmmiller** [comment]
New guide is out, feel free to have a look and offer up any suggestions:

https://github.com/mModule/guide_hma
